src/range.rs

```diff
@@ -54,6 +54,19 @@ pub fn TextRange(start: TextSize, end: TextSize) -> TextRange {
 
 /// Identity methods.
 impl TextRange {
+    /// Creates a zero-length range at the specified offset.
+    pub const fn empty(self, offset: TextSize) -> TextRange {
+        TextRange {
+            start: offset,
+            end: offset,
+        }
+    }
+
+    /// Creates a `TextRange` with specified starting offset and length.
+    pub fn offset_len(start: TextSize, length: TextSize) -> TextRange {
```

Comment:
I am not terribly happy about the naming here, but I do thing that we need a ctor with this signature. Doing `TextRange(start, start + len)` on the callsite doesn't work if `start` is a non-trivial expression.